contracts/SCBridgeWallet.sol

```diff
@@ -147,40 +147,58 @@ contract SCBridgeWallet is IAccount {
     }
   }
 
+  function validateUserOpOPEN(
+    UserOperation calldata userOp,
+    bytes32 userOpHash
+  ) internal view returns (uint256 validationData) {
+    // If the wallet is OPEN, require intermediary signature on every operation
+    bytes memory intermediarySig = userOp.signature[65:130];
+    return validateSignature(userOpHash, intermediarySig, intermediary);
+  }
+
+  function permitted(bytes4 functionSelector) internal pure returns (bool) {
+    return (functionSelector == this.challenge.selector ||
+      functionSelector == this.reclaim.selector ||
+      functionSelector == this.unlockHTLC.selector);
+  }
+
+  function validateUserOpCHALLENGE_RAISED(
+    UserOperation calldata userOp,
+    bytes32 // userOpHash
+  ) internal pure returns (uint256 validationData) {
+    // If the wallet is in CHALLENGE_RAISED, restrict operations by function selector
+    bytes4 functionSelector = bytes4(userOp.callData[0:4]);
+    require(permitted(functionSelector), "Invalid function selector");
+    return 0;
+  }
+
   function validateUserOp(
     UserOperation calldata userOp,
     bytes32 userOpHash,
     uint256 // missingAccountFunds
   ) external view returns (uint256 validationData) {
-    require(userOp.signature.length == 2 * 65, "Invalid signature length");
-    require(userOp.signature.length != 0x0, "Empty signature");
-
     bytes memory ownerSig = userOp.signature[0:65];
-    bytes memory intermediarySig = userOp.signature[65:130];
-
     // The owner of the wallet must always approve of any user operation to execute on it's behalf
     require(!isZero(ownerSig), "Must be signed by owner");
```

Comment:
There's actually a bug here, I thought this was doing a full check on the signature. But it isn't! I fixed this on #135

Reply:
Good spot! I also glanced over that line and assumed the same.